packages/block-library/src/navigation/style.scss

```
@@ -19,25 +24,199 @@
+		flex-direction: column;
+
+		// Allow modal to scroll.
+		overflow: auto;
+
+		// Give it a z-index just higher than the adminbar.
+		z-index: 100000;
+
+		padding: 24px;
+		background-color: inherit;
+
+		.wp-block-navigation__container {
+			display: flex;
+			flex-direction: column;
+			max-width: 400px;
+			margin-left: auto;
+			margin-right: auto;
+			align-items: flex-start;
+			line-height: 48px;
+			padding: 48px 0;
+
+			.wp-block-page-list {
+				flex-direction: column;
+			}
+		}
+	}
+
+	@include break-small() {
+		display: flex;
+		flex-direction: row;
+		position: relative;
+		background-color: inherit;
+
+		&.is-menu-open {
+			flex-direction: row;
+
+			.wp-block-navigation__container {
+				flex-direction: row;
+			}
+		}
+	}
+}
+
+// Default menu background and font color.
+.wp-block-navigation:not(.has-background) .wp-block-navigation__responsive-container {
+	background-color: #fff;
+	color: #000;
+}
+
+// Hide empty paragraph tags caused by wpautop.
+// @todo, need to not output them in the first place.
+.wp-block-navigation > p,
+.wp-block-navigation__responsive-container [data-micromodal-close] > p,
+.wp-block-navigation__responsive-container [role="dialog"] > p,
+.wp-block-navigation__responsive-container > p {
+	display: none;
+}
+
+// Menu and close buttons.
+.wp-block-navigation__responsive-container-open,
+.wp-block-navigation__responsive-container-close {
+	vertical-align: middle;
+	cursor: pointer;
+	color: currentColor;
+	background: transparent;
+	border: none;
+	margin: 0;
+	padding: 0;
+
+	svg {
+		fill: currentColor;
+		pointer-events: none;
+		display: block;
+		width: 24px;
+		height: 24px;
+	}
+}
+
+// Button to open the menu.
+.wp-block-navigation__responsive-container-open {
+	@include break-small {
+		display: none;
+	}
+}
+
+// Button to close the menus.
+.wp-block-navigation__responsive-container-close {
+	position: absolute;
+	top: 24px;
+	right: 24px;
+	z-index: 2; // Needs to be above the modal z index itself.
+
+	@include break-small() {
+		display: none;
+	}
+}
+
+// The menu adds a wrapping container.
+.wp-block-navigation__responsive-close,
+.wp-block-navigation__responsive-dialog,
+.wp-block-navigation__responsive-container-content {
+	width: 100%;
+	height: 100%;
+}
+
+// Always show submenus fully expanded inside the modal menu.
+.wp-block-navigation .wp-block-navigation__responsive-container.is-menu-open {
+	// This is clickable at the moment, though does not seem tabbable.
+	// @todo, verify that it doesnt' need a focus style.
+	&:focus {
+		outline: none;
+	}
+
+	.wp-block-navigation-link__submenu-icon {
+		display: none;
+	}
+
+	.has-child .submenu-container,
+	.has-child .wp-block-navigation-link__container {
+		position: relative;
+		opacity: 1;
+		visibility: visible;
+
+		padding: 0 0 0 32px;
+		border: none;
+	}
+
+	.wp-block-navigation-link,
+	.wp-block-pages-list__item {
+		flex-direction: column;
+		align-items: flex-start;
+	}
+}
+
+html.has-modal-open {
+	overflow: hidden;
+}
```

At the moment, we're not able to remove the `has-modal-open` class upon resizing past the small breakpoint, so we have to tell the class to only work within that boundary.

```suggestion
html.has-modal-open {
	overflow: hidden;
	@include break-small() {
		overflow: auto;
	}
}
```

---

I'd almost rather remove the feature again and revisit separately, than unset the rule at the breakpoint, to be honest. 

Last I checked, if you open the modal, resize to a big screen, the modal closes, but then resize back to the small screen, it opens again. So there's already something we need to deal with here. 

Could we keep the modal open as you resize? I.e. no matter _how_ you open the menu, if it's open it remains open regardless of size?

---

> Could we keep the modal open as you resize? I.e. no matter how you open the menu, if it's open it remains open regardless of size?

That's a great point, it would be very useful to have the user explicitly close it rather than going back to desktop on resize. Added in 6fba41b5581ebdafa13beec3e1b4ed195bc50f23.